### PR TITLE
[DO NOT MERGE] Challenges from training

### PR DESCRIPTION
**Pull request by cmtedbcosta**
- Sort the planets
- Retrieve planet specific details from API
- Conform design from Planet details screen to PDF Assignment

### CODE REVIEW

File: SwiftUITraining/SwiftUITraining/Data/PlanetService.swift [hunk 1]
```diff
@@ -20,8 +20,10 @@ class PlanetService : PlanetServiceProtocol{
         AF.request(baseUrl).responseDecodable(of: [Planet].self) { response in
             switch response.result {
                 case .success(let planets):
-                    closure(planets)
-                case .failure(let error):
+                    closure(planets.sorted {
+                        $0.distanceFromSun < $1.distanceFromSun
```
Comment (kateorlova) [inline SwiftUITraining/SwiftUITraining/Data/PlanetService.swift:25]: it's a good practice to transform a data into a presentation data in a class which is responsible for it
for example in `MVVM` it's a `viewModel`, in `MVP` it's a `presenter`
here we don't have neither `viewModel` nor `presenter`, so I'd say having sorting logic in a `view` would be a bit better than having it in a `service` which responsibility is only getting data from API

File: SwiftUITraining/SwiftUITraining/Views/PlanetDetailHeaderView.swift [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+//  SwiftUITraining
+//
+//  Created by Daniel Costa on 01/09/2020.
+//  Copyright © 2020 Coolblue. All rights reserved.
+//
+
+import SwiftUI
+import SDWebImageSwiftUI
+
+struct PlanetDetailHeaderView: View {
```
Comment (kateorlova) [inline SwiftUITraining/SwiftUITraining/Views/PlanetDetailHeaderView.swift:12]: nice that you created a separate `view` for a header 👍

File: SwiftUITraining/SwiftUITraining/Views/PlanetDetailRowView.swift [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    init(title: String, value: String?) {
+        self.title = title
+        self.stringValue = value ?? ""
+        self.doubleValue = 0
+    }
+    
+    init(title: String, value: Double?) {
+        self.title = title
+        self.doubleValue = value ?? 0
+        self.stringValue = ""
```
Comment (kateorlova) [inline SwiftUITraining/SwiftUITraining/Views/PlanetDetailRowView.swift:25]: would be better to have only `stringValue` and here in the initialiser transform incoming `doubleValue` to `stringValue`:
`stringValue = "\(doubleValue, specifier: "%.2f")"`
this way you can avoid code duplication below ⬇️

Comment (cmtedbcosta) [reply SwiftUITraining/SwiftUITraining/Views/PlanetDetailRowView.swift:25]: That's what I tried initially, but the conversion doesn't seem to be so straight forward...
image

File: SwiftUITraining/SwiftUITraining/Views/PlanetDetailHeaderView.swift (hunk 2 above)
Comment (cmtedbcosta) [reply SwiftUITraining/SwiftUITraining/Views/PlanetDetailHeaderView.swift:12]: =D

**Pull request closed without merging**